src/Servers/Kestrel/Core/src/Internal/Infrastructure/HttpUtilities.cs

```diff
@@ -430,20 +430,11 @@ internal static HttpVersion GetKnownVersionAndConfirmCR(this ReadOnlySpan<byte>
         /// <param name="knownScheme">A reference to the known scheme, if the input matches any</param>
         /// <returns>True when memory starts with known http or https schema</returns>
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
-        public static unsafe bool GetKnownHttpScheme(this Span<byte> span, out HttpScheme knownScheme)
+        public static bool GetKnownHttpScheme(this Span<byte> span, out HttpScheme knownScheme)
         {
-            fixed (byte* data = span)
+            if (span.Length >= sizeof(ulong))
```

Comment:
The JIT isn't always smart enough to match this manual check to the call to `BinaryPrimitives` below, so it could end up duplicating the bounds check. Suggest using `BinaryPrimitives.TryReadUInt64LittleEndian` instead, which guarantees that the bounds check will be emitted a maximum of once.

Reply:
Raised issue for it https://github.com/dotnet/runtime/issues/35348; will try the the `Try` version as you suggest